packages/web-components/fast-foundation/src/data-grid/data-grid-row.ts

```diff
@@ -0,0 +1,293 @@
+import {
+    attr,
+    FASTElement,
+    html,
+    RepeatBehavior,
+    RepeatDirective,
+    observable,
+    ViewTemplate,
+} from "@microsoft/fast-element";
+import {
+    keyCodeArrowLeft,
+    keyCodeArrowRight,
+    keyCodeEnd,
+    keyCodeHome,
+} from "@microsoft/fast-web-utilities";
+import { ColumnDefinition } from "./data-grid";
+
+/**
+ * Enumerates possible row types
+ *
+ * @public
+ * @beta
+ */
+export enum DataGridRowTypes {
+    default = "default",
+    header = "header",
+    stickyHeader = "sticky-header",
+}
+
+/**
+ * A Data Grid Row Custom HTML Element.
+ *
+ * @public
+ */
+export class DataGridRow extends FASTElement {
+    private static cellQueryString =
+        '[role="cell"], [role="gridcell"], [role="columnheader"]';
+
+    /**
+     * String that gets applied to the the css gridTemplateColumns attribute for the row
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: grid-template-columns
+     */
+    @attr({ attribute: "grid-template-columns" })
+    public gridTemplateColumns: string;
+    private gridTemplateColumnsChanged(): void {
+        if ((this as FASTElement).$fastController.isConnected) {
```

Comment:
same for all these `this as FASTElement` - doing the casting should not be necessary.